sentry/src/main/java/io/sentry/SentryExecutorService.java

```diff
@@ -2,43 +2,96 @@
 
 import io.sentry.util.AutoClosableReentrantLock;
 import java.util.concurrent.Callable;
-import java.util.concurrent.Executors;
+import java.util.concurrent.CancellationException;
 import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.jetbrains.annotations.TestOnly;
 
 @ApiStatus.Internal
 public final class SentryExecutorService implements ISentryExecutorService {
 
-  private final @NotNull ScheduledExecutorService executorService;
+  /**
+   * ScheduledThreadPoolExecutor grows work queue by 50% each time. With the initial capacity of 16
+   * it will have to resize 4 times to reach 40, which is a decent middle-ground for prewarming.
+   * This will prevent from growing in unexpected areas of the SDK.
+   */
+  private static final int INITIAL_QUEUE_SIZE = 40;
+
+  /**
+   * By default, the work queue is unbounded so it can grow as much as the memory allows. We want to
+   * limit it by 271 which would be x8 times growth from the default initial capacity.
+   */
+  private static final int MAX_QUEUE_SIZE = 271;
+
+  private final @NotNull ScheduledThreadPoolExecutor executorService;
   private final @NotNull AutoClosableReentrantLock lock = new AutoClosableReentrantLock();
 
+  @SuppressWarnings("UnnecessaryLambda")
+  private final @NotNull Runnable dummyRunnable = () -> {};
+
+  private final @Nullable SentryOptions options;
+
   @TestOnly
-  SentryExecutorService(final @NotNull ScheduledExecutorService executorService) {
+  SentryExecutorService(
```

Comment:
The core SDK, logging, spotlight and profiling all use their own instance of  `SentryExecutorService`. I guess it makes sense to pass the queue size as a constructor parameter instead and only prewarm the core SDK one?

Reply:
Oh I see, you only call pre-warm on the core one anyway.

Reply:
yeah, I think other usages are fine and shouldn't have many tasks scheduled